packages/benchmarks/src/implementations/styled-jsx/Box.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react/prop-types */
+import classnames from 'classnames';
+import React from 'react';
+import View from './View';
+
+const getColor = color => {
+  switch (color) {
+    case 0:
+      return '#14171A';
+    case 1:
+      return '#AAB8C2';
+    case 2:
+      return '#E6ECF0';
+    case 3:
+      return '#FFAD1F';
+    case 4:
+      return '#F45D22';
+    case 5:
+      return '#E0245E';
+    default:
+      return 'transparent';
+  }
+};
+
+const Box = props => {
+  const { className, children: styles } = (
+    <scope className={classnames('Box', props.fixed && 'fixed')}>
```

Comment:
What is `scope`?

Reply:
It is a dummy element, it could be any lowercase tag name e.g. `div`. This is a hack-workaround that a user came up with.

Because of the way how styled-jsx rewrites JSX, `scope` will get the scoped `className`: something like`jsx-someHash`. At runtime I extract it from the React Element together with `children` which are the actual styles or, to be precise, the styled-jsx React Component which takes care of [un]mounting and updating the actual CSS.

```jsx
const { props } = (
  <scope className="foo">
      <style jsx>{`.foo { color: red }`}</style>
  </scope>
)
```

is transpiled to

```jsx
const { props } = (
  <scope className="foo jsx-123">
      <_JSXStyle styleId={'123'} css={`.foo.jsx-123 { color: red }`} />
  </scope>
)
```

And finally to `React.createElement` calls.

Reply:
It is not bullet proof since passing down `foo` is not "safe", but imho it is ok when dealing with leaf elements.

Reply:
Thanks for the explanation